### ISSUE TITLE
Missed color for `CSI 2 m` with  the enabled support for pseudo consoles

### ISSUE THREAD

**Issue by AlttiRi** [git-for-windows/git#3823]
### I use
- Git-2.36.0-64-bit.exe
- Windows 10
- "MinTTY" option
- **"Enable experimental support for pseudo consoles." option**

### Bug
`\u001B[2m` does not makes text gray.

### More info

Python script:
```py
import shutil, sys

print(shutil.get_terminal_size())
sys.stdout.write("\u001B[2m"    +  "2m" + "\u001B[0m\n")
sys.stdout.write("\u001B[90m"   + "90m" + "\u001B[0m\n")
sys.stdout.write("\u001B[1;32m" + "32m" + "\u001B[0m\n")
sys.stdout.flush()
```

With enabled option "Enable experimental support for pseudo consoles." `"\u001B[2m"` console color is ignored:
![Screenshot](https://user-images.githubusercontent.com/16310547/165407173-d054dd68-48a7-419b-9077-cb294da4ed54.png)

Without this option (however, `get_terminal_size` does not work in this case):
![Screenshot_3](https://user-images.githubusercontent.com/16310547/165407190-598a39f7-0bd4-4cee-9e1f-0043e6b96d04.png)

Window Terminal:
![Screenshot_4](https://user-images.githubusercontent.com/16310547/165407208-ad9b99eb-9861-433a-904b-4454f554f515.png)

**Comment by dscho** [comment]
Does the same happen with [Cygwin](https://cygwin.com/)?

**Comment by AlttiRi** [comment]
Yes. It uses mintty 3.6.1.
![image](https://user-images.githubusercontent.com/16310547/165533323-21048482-0e34-4b54-9016-3e2493cce529.png)

**Comment by dscho** [comment]
I just integrated that new MinTTY version into Git for Windows' SDK.

Could you replace (as admin) `C:\Program Files\Git\usr\bin\mintty.exe` with [the version from the SDK](https://github.com/git-for-windows/git-sdk-64/blob/HEAD/usr/bin/mintty.exe) and see whether that addresses your problem?

**Comment by AlttiRi** [comment]
Nothing changed.

**Comment by dscho** [comment]
> So, it possibly should be addressed to MinTTY authors?

Potentially. However, I have to report that the problem does not reproduce for me (this is with MinTTY v3.6.0 and pseudo console support enabled):

![image](https://user-images.githubusercontent.com/127790/165622190-be889f8c-8937-4d05-9471-3a4bd74a8e9e.png)

**Comment by rimrul** [comment]
>However, I have to report that the problem does not reproduce for me (this is with MinTTY v3.6.0 and pseudo console support enabled):

Just to make sure, are you using native `python` through `winpty` or Msys2 `python`?

**Comment by AlttiRi** [comment]
**A thing made with Node.js works different that the same one made with Python.** (In Git-Bash terminal)

Node.js in both cases shows the console sizes, but it has very poor colors and it ignores `2`.

Python always shows proper colors, except for `2` with "pseudo consoles" option. 
The console size is shows only with "pseudo consoles" option.

In WT the results of Node.js and Python are the same, everything is proper.

Node.js code:
```js
const color = (code, text) => process.stdout.write("\u001B[" + code + "m" + text + "\u001B[0m\n");
color("34", "ANSI_BLUE-34");
color("1;31", "ANSI_RED_BOLD-1;31");
console.log("COMMON-TEXT; Console size: " + process.stdout.columns + " " + process.stdout.rows);
color("2", "TEXT-2 (should be gray)");
color("38;5;208", "8bit-ORANGE-208");
color("38;5;99", "8bit-VIOLET-99");
color("38;5;92", "8bit-VIOLET-92");
color("38;5;245", "8bit-GRAY-245");
```
Python code:
```py
import shutil, sys

def color(code, text):
    sys.stdout.write("\u001B[" + code + "m" + text + "\u001B[0m\n")

color("34", "ANSI_BLUE-34");
color("1;31", "ANSI_RED_BOLD-1;31");
print("COMMON-TEXT;", shutil.get_terminal_size());
color("2", "TEXT-2 (should be gray)");
color("38;5;208", "8bit-ORANGE-208");
color("38;5;99", "8bit-VIOLET-99");
color("38;5;92", "8bit-VIOLET-92");
color("38;5;245", "8bit-GRAY-245");
```
With "pseudo consoles":
![with](https://user-images.githubusercontent.com/16310547/165632523-17485b98-9220-4412-850d-30f465a9e051.png)

Without: 
![without](https://user-images.githubusercontent.com/16310547/165632556-d33ff77f-7423-4076-b3fc-78bbc51b544f.png)

Windows Terminal:

![wt](https://user-images.githubusercontent.com/16310547/165632593-56b2f0dd-8fb6-4e93-9aab-d64e103e49a1.png)

**Comment by dscho** [comment]
> > However, I have to report that the problem does not reproduce for me (this is with MinTTY v3.6.0 and pseudo console support enabled):
> 
> Just to make sure, are you using native `python` through `winpty` or Msys2 `python`?

I am running Python from the Microsoft Store, not via `winpty`.

> A thing made with Node.js works different that the same one made with Python

You're probably calling this with `node <script>.js`. However, `node` is a shell alias to `winpty node.exe`. If you run it via `node.exe`, I bet that it works the same as Python.

**Comment by AlttiRi** [comment]
Also can I toggle `Enable experimental support for pseudo consoles.` option without reinstalling the program?

**Comment by rimrul** [comment]
> Also can I toggle `Enable experimental support for pseudo consoles.` option without reinstalling the program?

Yes. Create `/etc/git-bash.config` with the content `MSYS=enable_pcon` to enable it, delete the file to disable it. And restart Git Bash after changing it.
